x-pack/platform/plugins/shared/streams_app/public/components/significant_events_discovery/components/streams_tree/tree_table.tsx

```diff
@@ -0,0 +1,376 @@
+            // Only show expand/collapse if tree mode is active and has children
+            const treeMode = shouldComposeTree(sortField);
+            const hasChildren = !!item.children && item.children.length > 0;
+            const isCollapsed = collapsed.has(item.stream.name);
+            return (
+              <EuiFlexGroup
+                alignItems="center"
+                gutterSize="s"
+                responsive={false}
+                className={css`
+                  margin-left: ${item.level * parseInt(euiTheme.size.xl, 10)}px;
+                `}
+              >
+                {treeMode && item.children && hasChildren && (
+                  <EuiFlexItem grow={false}>
+                    <EuiIcon
+                      type={isCollapsed ? 'arrowRight' : 'arrowDown'}
+                      color="text"
+                      size="m"
+                      data-test-subj={`${isCollapsed ? 'expand' : 'collapse'}Button-${
+                        item.stream.name
+                      }`}
+                      aria-label={i18n.translate(
+                        isCollapsed
+                          ? 'xpack.streams.streamsTreeTable.collapsedNodeAriaLabel'
+                          : 'xpack.streams.streamsTreeTable.expandedNodeAriaLabel',
+                        {
+                          defaultMessage: isCollapsed
+                            ? 'Collapsed node with {childCount} children'
+                            : 'Expanded node with {childCount} children',
+                          values: { childCount: item.children.length },
+                        }
+                      )}
+                      onClick={(e: React.MouseEvent) => {
+                        handleToggleCollapse(item.stream.name);
+                      }}
+                      tabIndex={0}
+                      role="button"
+                      onKeyDown={(e: React.KeyboardEvent) => {
+                        if (e.key === 'Enter' || e.key === ' ') {
+                          e.preventDefault();
+                          handleToggleCollapse(item.stream.name);
+                        }
+                      }}
+                      style={{ cursor: 'pointer' }}
+                    />
+                  </EuiFlexItem>
+                )}
+                {treeMode && !hasChildren && (
+                  <EuiFlexItem grow={false}>
+                    <EuiIcon type="empty" color="text" size="m" aria-hidden="true" />
+                  </EuiFlexItem>
+                )}
+                <EuiFlexItem grow={false}>
+                  <EuiLink
+                    data-test-subj={`streamsNameLink-${item.stream.name}`}
+                    href={router.link('/{key}', { path: { key: item.stream.name } })}
+                  >
+                    <EuiHighlight search={searchQuery?.text ?? ''}>{item.stream.name}</EuiHighlight>
+                  </EuiLink>
+                </EuiFlexItem>
+                {false && (
```

Comment:
Spinner is hidden for now until the table is integrated with the background tasks API.